ui/src/app/views/registration-dashboard.tsx

```diff
@@ -78,6 +81,7 @@ async function redirectToTraining() {
 export const RegistrationTasks = [
   {
     key: 'twoFactorAuth',
+    completionPropsKey: 'twoFactorAuthCompleted',
```

Comment:
This explicitly connects each registration task to its location within the props passed into this component.

(Note: in the longer term, we should probably refactor this component to directly refer to the user profile. As-is, there's a lot of boilerplate coordination between homepage and registration-dashboard that could probably be gutted. But that's a refactor for another time...)